miner.c

```diff
@@ -10508,8 +10505,11 @@ enum test_nonce2_result _test_nonce2(struct work *work, uint32_t nonce, bool che
 			{
 				applog(LOG_DEBUG, "Stratum pool %u target has changed since work job issued, checking that too",
 				       pool->pool_no);
-				if (hash_target_check_v(work->hash, pool->next_target))
+				if (hash_target_check_v(work->hash, pool->next_target)) {
 					high_hash = false;
+					memcpy(work->target, pool->next_target, sizeof(work->target));
+					calc_diff(work, 0);
```

Comment:
Okay, more input on this matter. My recent experience with Neptune shows that this logic (retroactive application of the newly set pool difficulty to old works) brings only troubles with well-behaving pools (most of Bitcoin pools are well-behaving). It only leads to excessive rejects, because normal pools expect those old works to satisfy the old difficulty, not the newly set one.
So, at least in Bitcoin case where we have "normal" pools I found that disabling that logic gave me greatly reduced rate of rejects. So, in the next Neptune firmware release, that will be disabled by default with a help of newly introduced command-line option. What do you think of adding that option to the mainline? The code is here:
https://github.com/KnCMiner/bfgminer/commit/bac9098cfc1a70a542bf14de739dea3ef1b69c37